c7n/resources/asg.py

```diff
@@ -1347,31 +1347,46 @@ class MarkForOp(Tag):
         key={'type': 'string'},
         tag={'type': 'string'},
         message={'type': 'string'},
-        days={'type': 'number', 'minimum': 0})
+        days={'type': 'number', 'minimum': 0},
+        hours={'type': 'number', 'minimum': 0})
 
     default_template = (
         'AutoScaleGroup does not meet org policy: {op}@{action_date}')
 
     def process(self, asgs):
+        self.tz = zoneinfo.gettz(
```

Comment:
this should move to a validate method, ideally we shouldn't even be in policy execution on a config validation error.

Reply:
I'm not sure if I quite follow, but I think I understand what you are saying. I think you are saying that by the time we get to process we should almost certain it's going to work, so there should be a validate method to check before we process. I see some examples elsewhere in the code base.

So I think you are asking that:

```
        if not self.tz:
            raise FilterValidationError(
                "Invalid timezone specified %s" % self.tz)
```
be moved into a `validate` method? 

I copied this logic from the previous PR that made the updated in `tags.py` seen here https://github.com/capitalone/cloud-custodian/blob/3a24d34bf59386f0377e984248e77aa28ccef7c5/c7n/tags.py#L608. But looking at the code again, I'm not sure if this `if not self.tz` is even needed as we set a value right above and default to `utc`. 

Maybe you are asking for:

```
        self.tz = zoneinfo.gettz(
            Time.TZ_ALIASES.get(self.data.get('tz', 'utc')))
```

to be moved out of the `process` method and then the `if not self.tz` moved to a `validate` method?

Any clarification here would be helpful and I'll apply the same to the code in `tags.py`.